.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+name: Publish a Release
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version without leading "v" (1.0, 2.3.4, 0.1.0-pre1)'
+        required: true
+        default: ''
+      image_tags:
+        description: 'Extra docker image tags ("latest,test")'
+        required: true
+        default: 'latest,release'
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    permissions:
+      contents: write
+      packages: write
+    steps:
+    - name: Ensure version does not start with 'v'
+      uses: actions/github-script@v7
+      with:
+        github-token: ${{ github.token }}
+        script: |
+          version = context.payload.inputs.version;
+          if (/^v/.test(version)) throw new Error("Bad version number: " + version)
+    - uses: actions/checkout@v4
+    - name: Cache data/sources
+      uses: ./.github/cache-sources-action
+    - uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'temurin'
+        cache: 'maven'
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
```

Comment:
you shouldn't need these server settings if you aren't publishing to maven central

Reply:
Removed.